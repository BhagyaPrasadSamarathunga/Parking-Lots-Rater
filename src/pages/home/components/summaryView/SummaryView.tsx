import React, { useState } from 'react';

interface ParkingLot {
  id: string;
  name: string;
  address: string;
  image: string;
}

interface SummaryViewProps {
  goodParkingLots: ParkingLot[];
  badParkingLots: ParkingLot[];
}

const ParkingLotItem = ({item}: {item:ParkingLot}) => {
  return (
    <div className='flex flex-col w-30 h-30 m-2 '>
      <img className='flex w-20 h-20 place-self-center' src={item.image} />
        <span className='text-xs font-semibold'>{item.name}</span>
        <span className='text-xs max-w-20 text-wrap font-extralight'>{item.address}</span>
    </div>
  )
}

const SummaryView: React.FC<SummaryViewProps> = ({ goodParkingLots, badParkingLots }) => {
  const [filter, setFilter] = useState<string>('');
  const [sortBy, setSortBy] = useState<string>('name');

  const filterAndSortLots = (lots: ParkingLot[]) => {
    return lots
      .filter((lot) => lot.name.toLowerCase().includes(filter.toLowerCase()) || lot.address.toLowerCase().includes(filter.toLowerCase()))
      .sort((a, b) => a[sortBy].localeCompare(b[sortBy]));
  };

  const filteredGoodParkingLots = filterAndSortLots(goodParkingLots);
  const filteredBadParkingLots = filterAndSortLots(badParkingLots);

  return (
    <div className='flex flex-col w-3/4 justify-center items-center'>
      <div className="mb-4 ">
        <input
          type="text"
          placeholder="Filter by name or address"
          value={filter}
          onChange={(e) => setFilter(e.target.value)}
          className="px-2 py-1 border rounded"
        />
        <select value={sortBy} onChange={(e) => setSortBy(e.target.value)} className="ml-2 px-2 py-1 border rounded">
          <option value="name">Name</option>
          <option value="address">Address</option>
        </select>
      </div>
      <h2 className='w-full justify-center flex animate-pulse text-stone-800'>Good parking Lots</h2>
      <div className='flex flex-row overflow-x-auto overflow-x-scroll'>
        {filteredGoodParkingLots.map((lot) => (
          <ParkingLotItem key={lot.id} item={lot} />
        ))}
      </div>
      <h2 className='w-full justify-center flex animate-pulse text-stone-800'>Bad Parking Lots</h2>
      <div className='flex flex-row'>
        {filteredBadParkingLots.map((lot) => (
           <ParkingLotItem key={lot.id} item={lot} />
        ))}
      </div>
    </div>
  );
};

export default SummaryView;