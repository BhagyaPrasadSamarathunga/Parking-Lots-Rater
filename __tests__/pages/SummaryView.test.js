import React from 'react';
import { render } from '@testing-library/react';
import SummaryView from '../../src/pages/home/components/summaryView/SummaryView';

const goodLotsList = [ 
    {
        "id": "0945f12b-5b44-428d-aec5-df703dfadeb6",
        "image": "https://images.freeimages.com/images/large-previews/e8e/underground-parking-1206464.jpg",
        "live_date": "2021-01-01",
        "name": "Lot Munich 29",
        "size": 25,
        "address": "Verdistrasse 209, 81257 München",
        "__typename": "Lot"
    },
    {
        "id": "0ef7ff09-b32e-43e6-ab58-95b6c0a39b4a",
        "image": "https://images.freeimages.com/images/large-previews/e8e/underground-parking-1206464.jpg",
        "live_date": "2020-11-03",
        "name": "Lot Munich 26",
        "size": 25,
        "address": "Verdistrasse 26, 81257 München",
        "__typename": "Lot"
    },];
const badLotsList = [
    {
        "id": "15800269-f4ac-4e83-9bfe-e3da470075d0",
        "image": "https://images.freeimages.com/images/large-previews/68d/narrow-street-1204865.jpg",
        "live_date": "2021-06-01",
        "name": "Lot Munich 30",
        "size": 25,
        "address": "Amalienburgstr. 130, 81247 München",
        "__typename": "Lot"
    },
    {
        "id": "19d9fb52-d0b3-43b0-8c38-1a4a8b9caa5b",
        "image": "https://images.freeimages.com/images/large-previews/e8e/underground-parking-1206464.jpg",
        "live_date": "2020-10-04",
        "name": "Lot Munich 24",
        "size": 25,
        "address": "Verdistrasse 90, 81257 München",
        "__typename": "Lot"
    },
    {
        "id": "2ba3887e-8db6-4173-8e48-d76cccfa8454",
        "image": "https://images.freeimages.com/images/large-previews/e8e/underground-parking-1206464.jpg",
        "live_date": null,
        "name": "Lot Munich 36",
        "size": 25,
        "address": "Verdistrasse 146 81257 München",
        "__typename": "Lot"
    }];
    
describe('Summary View component', () => {
    test('matches snapshot', () => {
      const { asFragment } = render(<SummaryView goodParkingLots={goodLotsList} badParkingLots ={badLotsList}/>);
      expect(asFragment()).toMatchSnapshot();
    });
  });