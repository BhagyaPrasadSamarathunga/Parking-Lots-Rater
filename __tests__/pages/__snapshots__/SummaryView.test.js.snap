// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary View component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col w-3/4 justify-center items-center"
  >
    <div
      class="mb-4 "
    >
      <input
        class="px-2 py-1 border rounded"
        placeholder="Filter by name or address"
        type="text"
        value=""
      />
      <select
        class="ml-2 px-2 py-1 border rounded"
      >
        <option
          value="name"
        >
          Name
        </option>
        <option
          value="address"
        >
          Address
        </option>
      </select>
    </div>
    <h2
      class="w-full justify-center flex animate-pulse text-stone-800"
    >
      Good parking Lots
    </h2>
    <div
      class="flex flex-row overflow-x-auto overflow-x-scroll"
    >
      <div
        class="flex flex-col w-30 h-30 m-2 "
      >
        <img
          class="flex w-20 h-20 place-self-center"
          src="https://images.freeimages.com/images/large-previews/e8e/underground-parking-1206464.jpg"
        />
        <span
          class="text-xs font-semibold"
        >
          Lot Munich 26
        </span>
        <span
          class="text-xs max-w-20 text-wrap font-extralight"
        >
          Verdistrasse 26, 81257 München
        </span>
      </div>
      <div
        class="flex flex-col w-30 h-30 m-2 "
      >
        <img
          class="flex w-20 h-20 place-self-center"
          src="https://images.freeimages.com/images/large-previews/e8e/underground-parking-1206464.jpg"
        />
        <span
          class="text-xs font-semibold"
        >
          Lot Munich 29
        </span>
        <span
          class="text-xs max-w-20 text-wrap font-extralight"
        >
          Verdistrasse 209, 81257 München
        </span>
      </div>
    </div>
    <h2
      class="w-full justify-center flex animate-pulse text-stone-800"
    >
      Bad Parking Lots
    </h2>
    <div
      class="flex flex-row"
    >
      <div
        class="flex flex-col w-30 h-30 m-2 "
      >
        <img
          class="flex w-20 h-20 place-self-center"
          src="https://images.freeimages.com/images/large-previews/e8e/underground-parking-1206464.jpg"
        />
        <span
          class="text-xs font-semibold"
        >
          Lot Munich 24
        </span>
        <span
          class="text-xs max-w-20 text-wrap font-extralight"
        >
          Verdistrasse 90, 81257 München
        </span>
      </div>
      <div
        class="flex flex-col w-30 h-30 m-2 "
      >
        <img
          class="flex w-20 h-20 place-self-center"
          src="https://images.freeimages.com/images/large-previews/68d/narrow-street-1204865.jpg"
        />
        <span
          class="text-xs font-semibold"
        >
          Lot Munich 30
        </span>
        <span
          class="text-xs max-w-20 text-wrap font-extralight"
        >
          Amalienburgstr. 130, 81247 München
        </span>
      </div>
      <div
        class="flex flex-col w-30 h-30 m-2 "
      >
        <img
          class="flex w-20 h-20 place-self-center"
          src="https://images.freeimages.com/images/large-previews/e8e/underground-parking-1206464.jpg"
        />
        <span
          class="text-xs font-semibold"
        >
          Lot Munich 36
        </span>
        <span
          class="text-xs max-w-20 text-wrap font-extralight"
        >
          Verdistrasse 146 81257 München
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
